core/src/main/scala/filodb.core/binaryrecord2/RecordSchema.scala

```diff
@@ -289,6 +289,47 @@ final class RecordSchema(val columns: Seq[ColumnInfo],
     result
   }
 
+
+  /**
+   * Reads all the label key-value pairs from the BinaryRecord and creates a map.
+   * If multiple labels with same key --> then the last writer wins.
+   * @param base data source
+   * @param offset offset to read from
+   * @return Map of all the tags/labels. key is label-name, value is label-value
+   */
+  def toStringPairsMap(base: Any, offset: Long): mutable.Map[String, String] = {
```

Comment:
I don't actually, but wanted to keep it consistent with `toStringPairs` but also agree that it is wasteful .. should I rename it to `getStringValues` and only fetch for columnTypes `StringColumn` and `MapColumn` ? ( since I only need those in downsample cardinality .. on the other hand, we use it when going over the partKeys, which only contain the tags and metric name)

Reply:
@alextheimer let me know your thoughts ?

Reply:
Initially I did that, but then I wanted to avoid two memory allocations. First creation of array and then convert to map ( not sure if Scala does optimization underneath to avoid it, but since they were two separate data-structures, I thought this would be to separate allocations )